rotkehlchen/rotkehlchen.py

```diff
@@ -1245,6 +1247,48 @@ def query_balances(
         self.task_manager.should_schedule = True  # type: ignore[union-attr]  # should exist here
         return result_dict
 
+    def _save_asset_prices_as_historical(self, result_dict: dict[str, Any]) -> None:
+        """Query current prices for assets in result_dict and save as historical prices
+
+        Fetches prices in the user's main currency since that's how they'll be requested
+        for by the graphs.
+        """
+        assets_to_query: set[Asset] = set()
+        for assets_dict in (result_dict['assets'], result_dict['liabilities']):
+            assets_to_query.update(assets_dict)
+
+        if len(assets_to_query) == 0:
+            return
+
+        with self.data.db.conn.read_ctx() as cursor:
+            main_currency = self.data.db.get_setting(cursor, 'main_currency')
+
+        # filter out main currency from assets to query
+        if len(assets_to_price := [asset for asset in assets_to_query if asset != main_currency]) == 0:  # noqa: E501
+            return
+
+        historical_prices, current_ts = [], ts_now()
+        try:
+            for asset, (price, oracle) in Inquirer.find_prices_and_oracles(
```

Comment:
why do this logic additionally? Isn't this price queried anyway to populate the `result_dict` to pass here? Aren't you essentially repeating the entire price query loop for all assets every time with this function?